elixir/apps/web/lib/web/live/sites/new_token.ex

```diff
@@ -253,19 +253,18 @@ defmodule Web.Sites.NewToken do
         chmod +x /usr/local/bin/firezone-gateway; \\
       fi \\
     '
-    ExecStartPre=/usr/bin/mkdir -p /etc/firezone
-    ExecStartPre=/usr/bin/chown firezone:firezone /etc/firezone
-    ExecStartPre=/usr/bin/chmod 0755 /etc/firezone
-    ExecStartPre=/usr/bin/chmod +x /usr/local/bin/firezone-gateway
+    ExecStartPre=/bin/sh -c 'mkdir -p /etc/firezone'
+    ExecStartPre=/bin/sh -c 'chown firezone:firezone /etc/firezone'
```

Comment:
If the mkdir above fails, won't this `chown` fail as well?

Reply:
Is the /etc directory needed? I've been communicating to pilot users that the gateway doesn't require a persistent disk and can be configured through ENV vars, so maybe we make this optional?

Reply:
What I'm planning to do for documentation is to show a barebones basic unit file that the more savvy users can use as a starting point, but also have this auto-updating version as the default (i.e. preferred) version.

Reply:
@jamilbk there is no step here that will fail silently, but systemd will run these commands as root so unless `/etc` is mounted as a read-only volume we should be always able to write to it. In terms of documentation - it's completely optionally for the gateway, but it's the most common place to store our internal data.

> What I'm planning to do for documentation is to show a barebones basic unit file that the more savvy users can use as a template / starting point, but also have this auto-updating version as the default (i.e. preferred) version.

I would recommend not to give a barebone unitfile for people who are not very savvy in the documentation, this is a pretty complex way to run the gateway, and really hard to debug one. I think it will cause more pain than good. Plus there is no way even the most advanced users can know which capabilities, temporary directories, file permissions, etc our gateway needs.